src/vs/base/common/cancellation.ts

```diff
@@ -93,6 +99,8 @@ export class CancellationTokenSource {
 	}
 
 	dispose(): void {
-		this.cancel();
+		if (this._token instanceof MutableToken) {
```

Comment:
@jrieken unclear to me what should happen if `dispose` is called without token being there. It almost seems like we should have a new `CancellationToken.Disposed` that is a) not canceled and b) has no-ops for the methods.

Alternatively to keep current semantics we could say that in this case we do `this._token = CancellationToken.Cancelled`